File: nodejs/src/sinDependencyInjection/UserController.js
import {UserService} from "./UserService";

export class UserControllerSinDI {

    userService = new UserService();


    create(ctx) {
        const {nombre, apellido} = ctx.req.query;
        const user = this.userService.create(nombre, apellido);
        ctx.res.json(user);
    }

    findById(ctx) {
        const {id} = ctx.req.query;
        const user = this.userService.findById(id)
        ctx.res.json(user);
    }
}

File: kotlin/src/main/kotlin/com/di/kotlin/no/di/UserController.kt
package com.di.kotlin.no.di

import com.di.kotlin.model.User
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping

@Controller("/no-di")
class UserController {

    val userService = UserService()

    @PostMapping()
    fun getById(nombre: String, apellido: String): User {
        return this.userService.create(nombre, apellido)
    }
}